vllm/entrypoints/openai/serving_completion.py

```diff
@@ -267,16 +267,18 @@ async def completion_stream_generator(
                     previous_num_tokens[i] = len(output.token_ids)
                     finish_reason = output.finish_reason
                     stop_reason = output.stop_reason
-                    if output.finish_reason is not None:  # return final usage
+                    if (request.stream_options
+                            and request.stream_options.continuous_usage_stats
+                            or (output.finish_reason is not None)):
```

Comment:
I think for consistency this should be:
```python
if (request.stream_options
        and request.stream_options.include_usage
        and request.stream_options.continuous_usage_stats
        or (output.finish_reason is not None)):
 ```

Reply:
The original if statement was  `if output.finish_reason is not None:`
If true, then the actual usage statistics were assigned to `current_usage` (this was before called `final_usage`), regardless of the value of `include_usage`. The if statement at lines 296 and following, will determine if this should be used or not to report these statistics for each server reply and the if statement at line 306 is used to determine if this should be used or not in `final_usage_chunk`.

Again, I tried to keep the code as close as possible to the original before the modification. In this case, it means that "include_usage" does not to be tested here, but that this will be done later.

I actually tested all those cases (all combinations), and looked at the server responses. These are all correct. This is also covered by the test that I added.

I do agree, however, that there is redundancy in the original code in the way that the conditions etc. are implemented at lines 276 and 290 and following. We can of course, remove that redundancy by adapting the conditions in the if statements as you suggested.